app/src/main/java/org/groundplatform/android/ui/datacollection/DataCollectionViewModel.kt

```diff
@@ -197,23 +197,23 @@ internal constructor(
   fun getTaskViewModel(taskId: String): AbstractTaskViewModel? {
     val viewModels = taskViewModels.value
 
-    val task = tasks.first { it.id == taskId }
-
-    if (viewModels.containsKey(taskId)) {
-      return viewModels[taskId]
-    }
```

Comment:
Can this be moved outside of the modified logic since it is not dependent upon the `task` variable?